### PR TITLE
feat(python): .to_physical() for List(Categorical)

### PR DESCRIPTION
**Pull request by jonashaag**
Fixes https://github.com/pola-rs/polars/issues/7655

### CODE REVIEW

File: polars/polars-core/src/series/mod.rs [hunk 1]
```diff
@@ -410,6 +411,11 @@ impl Series {
             Datetime(_, _) | Duration(_) | Time => Cow::Owned(self.cast(&Int64).unwrap()),
             #[cfg(feature = "dtype-categorical")]
             Categorical(_) => Cow::Owned(self.cast(&UInt32).unwrap()),
+            #[cfg(feature = "dtype-categorical")]
+            List(inner) => match &**inner {
```
Comment (ritchie46) [inline polars/polars-core/src/series/mod.rs:415]: I think we can make this more generic by calling `cast(dtype.to_physical())`.

File: py-polars/polars/expr/expr.py [hunk 2]
```diff
@@ -247,6 +247,7 @@ def to_physical(self) -> Self:
         - :func:`polars.datatypes.Time` -> :func:`polars.datatypes.Int64`
         - :func:`polars.datatypes.Duration` -> :func:`polars.datatypes.Int64`
         - :func:`polars.datatypes.Categorical` -> :func:`polars.datatypes.UInt32`
+        - ``List(Categorical)`` -> ``List(UInt32)``
```
Comment (ritchie46) [inline py-polars/polars/expr/expr.py:250]: This should be physical for all nested types. I think we should mention that all the inner type is represented by the physical types stated above.

File: py-polars/polars/series/series.py [hunk 3]
```diff
@@ -2992,6 +2992,7 @@ def to_physical(self) -> Series:
         - :func:`polars.datatypes.Time` -> :func:`polars.datatypes.Int64`
         - :func:`polars.datatypes.Duration` -> :func:`polars.datatypes.Int64`
         - :func:`polars.datatypes.Categorical` -> :func:`polars.datatypes.UInt32`
+        - ``List(Categorical)`` -> ``List(UInt32)``
```
Comment (ritchie46) [inline py-polars/polars/series/series.py:2995]: This is too specific. We could just mention that physical types are converted recursively.

**Comment by ritchie46** [comment]
@jonashaag It is almost there. Could you update or remove that specific docstring line?

**Comment by jonashaag** [comment]
@ritchie46 thanks, missed that one

**Comment by jonashaag** [comment]
@ritchie46 i think the test failure is unrelated

**Comment by ritchie46** [comment]
Thanks @jonashaag!

**Pull request merged**